cmd/scanner-trivy/main.go

```diff
@@ -33,67 +33,39 @@ func main() {
 		"built_at": date,
 	}).Info("Starting harbor-scanner-trivy")
 
-	jobQueueConfig, err := etc.GetJobQueueConfig()
+	config, err := etc.GetConfig()
 	if err != nil {
 		log.Fatalf("Error: %v", err)
 	}
-	log.Debugf("Job queue config %v", jobQueueConfig)
 
-	worker := queue.NewWorker(jobQueueConfig)
-
-	apiConfig, err := etc.GetAPIConfig()
-	if err != nil {
-		log.Fatalf("Error: %v", err)
-	}
-	log.Debugf("API server config: %v", apiConfig)
-
-	apiHandler, err := newAPIHandler(jobQueueConfig)
-	if err != nil {
-		log.Fatalf("Error: %v", err)
-	}
-
-	server := api.NewServer(apiConfig, apiHandler)
+	worker := queue.NewWorker(config.JobQueue)
+	apiServer := newAPIServer(config)
+	metricsServer := metrics.NewServer(config.Metrics)
 
 	shutdownComplete := make(chan struct{})
 	go func() {
 		sigint := make(chan os.Signal, 1)
-		signal.Notify(sigint, os.Interrupt, syscall.SIGTERM)
+		signal.Notify(sigint, syscall.SIGINT, syscall.SIGTERM)
 		captured := <-sigint
 		log.WithField("signal", captured.String()).Debug("Trapped os signal")
 
-		log.Debug("API server shutdown started")
-		if err := server.Shutdown(context.Background()); err != nil {
-			log.WithError(err).Error("Error while shutting down server")
-		}
-		log.Debug("API server shutdown completed")
-
-		log.Debug("Job queue shutdown started")
+		apiServer.Shutdown(context.Background())
+		metricsServer.Shutdown(context.Background())
```

Comment:
I'm curious about why the metrics server is launched separately. How about adding the handler for metrics to APIHandler? I think that two servers would use a lot of resources.